src/Authentication/Authentication/test/Find-MgGraphPermission.Tests.ps1

```diff
@@ -2,15 +2,170 @@
+        }
     }
 
-    Context "when executing the command" {
+    Context "When executing the command using a constrained set of permissions returned by MS Graph" {
+        BeforeAll {
+            Mock Invoke-MgGraphRequest {$permissionData}
+        }
+
+        It 'Executes successfully with no parameters' {
+            { Find-MgGraphPermission | Out-Null } | Should -Not -Throw
+            Assert-MockCalled Invoke-MgGraphRequest
+        }
+
+        It "Retrieves the expected set of delegated and app-only permissions when a search string is specified" {
+            { Find-MgGraphPermission mail | Out-Null } | Should -Not -Throw
+            Assert-MockCalled Invoke-MgGraphRequest
+        }
+
+        It "Returns nothing and throws no exception if a search string is specified and there is no match" {
+            { Find-MgGraphPermission 'Microsoft is the greatest company' | Out-Null } | Should -Not -Throw
+            Assert-MockCalled Invoke-MgGraphRequest
+        }
+    }
+
+    Context "When executing the command without a connection to MS Graph" {
+        BeforeAll {
+            Mock Invoke-MgGraphRequest {
+
+                Throw [System.Management.Automation.ValidationMetadataException]::new('mock connection error message')
+
+            }
+        }
+
+        It 'Executes successfully with no parameters' {
+            { Find-MgGraphPermission | Out-Null } | Should -Not -Throw
+            Assert-MockCalled Invoke-MgGraphRequest
+        }
+
+        It "Retrieves the expected set of delegated and app-only permissions when a search string is specified" {
+            { Find-MgGraphPermission mail | Out-Null } | Should -Not -Throw
+            Assert-MockCalled Invoke-MgGraphRequest
+        }
+
+        It "Returns nothing and throws no exception if a search string is specified and there is no match" {
+            { Find-MgGraphPermission 'Microsoft is the greatest company' | Out-Null } | Should -Not -Throw
+            Assert-MockCalled Invoke-MgGraphRequest
+        }
+    }
+
+    Context "When executing the command after initially not having a connection to MS Graph but getting it later" {
+
+        BeforeEach {
+            Mock Invoke-MgGraphRequest {
+
+                Throw [System.Management.Automation.ValidationMetadataException]::new('mock connection error message')
+
+            }
+        }
+
         It 'Executes successfully with no parameters' {
             { Find-MgGraphPermission | Out-Null } | Should -Not -Throw
+            Assert-MockCalled Invoke-MgGraphRequest
+
+            Mock Invoke-MgGraphRequest{
+                $permissionData
+            }
+
+            { Find-MgGraphPermission | Out-Null } | Should -Not -Throw
+            Assert-MockCalled Invoke-MgGraphRequest
+
+        }
+
+        It "Retrieves the expected set of delegated and app-only permissions when a search string is specified" {
+            { Find-MgGraphPermission mail | Out-Null } | Should -Not -Throw
```

Comment:
Rather than check to see if it doesn't throw, just do `$permissions  Find-MgGraphPermission mail`. Then your next line can be `$permissions.length | Should -Be 4` (or whatever is the number of permissions in the file). And add more lnes like `$permissions.ConsentType | Should -Be Admin` or whatever the right value is. You'll want to pick useful cases and validate that the code is behaving as you expect.